ckanext/cloudstorage/storage.py

```diff
@@ -191,13 +191,34 @@ def upload(self, id, max_size=10):
         :param max_size: Ignored.
         """
         if self.filename:
-            self.container.upload_object_via_stream(
-                self.file_upload,
-                object_name=self.path_from_filename(
-                    id,
-                    self.filename
+            if self.can_use_advanced_azure:
+                from azure.storage import blob as azure_blob
+
+                blob_service = azure_blob.BlockBlobService(
+                    self.driver_options['key'],
+                    self.driver_options['secret']
+                )
+
+                return blob_service.create_blob_from_stream(
+                    container_name=self.container_name,
+                    blob_name=self.path_from_filename(
+                        id,
+                        self.filename
+                        ),
```

Comment:
Your closing brackets should be on the same indent level as they opened, please.